web/pages/profile.tsx

```diff
@@ -525,6 +513,48 @@ function Profile() {
       </Box>
     </Flex>
   );
+
+  function EditButton() {
+    return !editing ? (
+      <Button
+        variant="solid-primary"
+        onClick={() => setEditing(true)}
+        width={"100%"}
+      >
+        Edit
+      </Button>
+    ) : (
+      <Stack direction="row">
+        <Button
+          variant="outline-secondary"
+          fontWeight="thin"
+          borderWidth="thin"
+          width={["50%", "100%"]}
+          onClick={() => {
+            dispatch({ type: "clear" });
+            setEditing(false);
+          }}
+        >
+          Cancel
+        </Button>
+        <Button
+          variant="solid-primary"
+          width={["50%", "100%"]}
+          onClick={async () => {
+            const req = await updatePreferences.mutateAsync({
+              uid: userData!.uid,
+              updateFields: preferences,
+            });
+            if (req.success) {
+              setEditing(false);
+            }
```

Comment:
Optional: you could add logic to display a toast if the request isn't successful. This is out of scope for your ticket, so no need to worry about it!